src/helper.cc

```diff
@@ -1045,12 +1045,12 @@ helperHandleRead(const Comm::ConnectionPointer &conn, char *, size_t len, Comm::
 
     if (!srv->stats.pending && !srv->stats.timedout) {
         /* someone spoke without being spoken to */
-        debugs(84, DBG_IMPORTANT, "ERROR: helperHandleRead: unexpected read from " <<
+        debugs(84, DBG_IMPORTANT, "ERROR: Killing helper process after an unexpected read from " <<
                hlp->id_name << " #" << srv->index << ", " << (int)len <<
                " bytes '" << srv->rbuf << "'");
 
-        srv->roffset = 0;
-        srv->rbuf[0] = '\0';
```

Comment:
This `srv` state adjustment may be reasonably viewed as a safety feature -- do not leave tainted input in the buffer. Yes, we are going to destroy `srv` very soon, but a lot of things may happen before that destruction. If we are removing this code, then we should lower the `srv->rbuf` adjustment above -- to keep `srv->buf` as is. However, that code move would kill fairly critical input debugging for this use case!

Please keep this code instead.

Same for the stateful helper, of course.

Reply:
Done, but note that there are some inconsistencies with such cleanup in other closePipesSafely() callers. E.g., helperStatefulHandleRead() zeroes only srv->roffset and some callers do nothing.